db.h

```diff
@@ -373,16 +383,22 @@ class CAddrDb
                             }
                         });)
 
-    void Add(const CAddress& addr, bool fForce = false)
+    void Add(const CAddress& addr, const CServiceResult& result, bool fForce = false)
     {
         CRITICAL_BLOCK(cs)
-        Add_(addr, fForce);
+        Add_(addr, result, fForce);
     }
-    void Add(const std::vector<CAddress>& vAddr, bool fForce = false)
+    void Add(const std::vector<CAddress>& vAddr, const std::vector<CServiceResult>& result, bool fForce = false)
     {
         CRITICAL_BLOCK(cs)
         for (int i = 0; i < vAddr.size(); i++)
-            Add_(vAddr[i], fForce);
+            Add_(vAddr[i], result[i], fForce);
+    }
+    void Add(const std::vector<CAddress>& vAddr,  bool fForce = false)
+    {
+        CRITICAL_BLOCK(cs)
+        for (int i = 0; i < vAddr.size(); i++)
+            Add_(vAddr[i], CServiceResult(), fForce);
```

Comment:
where `CServiceResult()` is defined? I see the struct defined in `db.h` but no function with that name.

Reply:
It executes the default constructor

Reply:
Ouch.Dumb question, isn't it? :smile: